closeio_api/__init__.py

```diff
@@ -51,18 +51,18 @@ def get(self, endpoint, data=None):
             for k, v in data.iteritems():
                 data[k] = unicode(v).encode('utf-8')
             endpoint += '/?'+urllib.urlencode(data)
-        else:
+        elif not endpoint.endswith('/'):
             endpoint += '/'
         return self.dispatch('get', endpoint)
 
     def post(self, endpoint, data):
-        return self.dispatch('post', endpoint+'/', data)
+        return self.dispatch('post', endpoint+('' if endpoint.endswith('/') else '/'), data)
 
     def put(self, endpoint, data):
-        return self.dispatch('put', endpoint+'/', data)
+        return self.dispatch('put', endpoint+('' if endpoint.endswith('/') else '/'), data)
 
     def delete(self, endpoint):
-        return self.dispatch('delete', endpoint+'/')
+        return self.dispatch('delete', endpoint+('' if endpoint.endswith('/') else '/'))
```

Comment:
hm. anyway we will keep one check in the get method, because may be query string.
I'll look, maybe better to do it with __call__ and __getattr__ . I mean one method for get/post/put/delete